src/asm/math.c

```diff
@@ -37,12 +38,15 @@ void math_DefinePI(void)
  */
 void math_Print(int32_t i)
 {
-	if (i >= 0)
-		printf("%d.%05d", i >> 16,
-		       ((int32_t)(fx2double(i) * 100000 + 0.5)) % 100000);
-	else
-		printf("-%d.%05d", (-i) >> 16,
-		       ((int32_t)(fx2double(-i) * 100000 + 0.5)) % 100000);
+	const char *sign = "";
+
+	if (i < 0) {
+		i = -i;
```

Comment:
Consider the case of `i` = `INT32_MIN` on two's complement platforms. The correct solution would be to use an unsigned variable to hold the absolute value.

Reply:
It wasn't checked for before, either. I might be able to check for this but I'm unsure if it's likely (or possible).